README.md

```diff
@@ -1,707 +1,1799 @@
+
+| Name   | Description                  |
+| :----- | :--------------------------- |
+| `func` | The function to be memoized. |
+
+</details>
+
+<details><summary><b>Returns</b></summary>
+
+A memoized function with a `.clearCache()` method attached.
+
+</details>
+
+<details><summary><b>Type parameters</b></summary>
+
+| Name   | Description                                |
+| :----- | :----------------------------------------- |
+| `Func` | The type of the function that is memoized. |
+
+</details>
+
+<details><summary><b><code>Examples</code></b></summary>
+
+Prior to `weakMapMemoize`, you had this problem:
+
+```ts
+const parametricSelector = createSelector(
+  [(state: RootState) => state.todos, (state: RootState, id: number) => id],
+  (todos, id) => todos.filter(todo => todo.id === id)
+)
+
+parametricSelector(state, 0) // Selector runs
+parametricSelector(state, 0)
+parametricSelector(state, 1) // Selector runs
+parametricSelector(state, 0) // Selector runs again!
+```
+
+Before you could solve this in a number of different ways:
+
+1. Set the `maxSize` with [`defaultMemoize`]:
+
+```ts
+const parametricSelector = createSelector(
+  [(state: RootState) => state.todos, (state: RootState, id: number) => id],
+  (todos, id) => todos.filter(todo => todo.id === id),
+  {
+    memoizeOptions: {
+      maxSize: 10
+    }
+  }
+)
+```
+
+But this required having to know the cache size ahead of time.
+
+2. Create unique selector instances using [`useMemo`].
+
+```tsx
+const parametricSelector = (id: number) =>
+  createSelector([(state: RootState) => state.todos], todos =>
+    todos.filter(todo => todo.id === id)
+  )
+
+const MyComponent: FC<Props> = ({ id }) => {
+  const selectTodosById = useMemo(() => parametricSelector(id), [id])
+
+  const todosById = useSelector(selectTodosById)
+
+  return (
+    <div>
+      {todosById.map(todo => (
+        <div key={todo.id}>{todo.title}</div>
+      ))}
+    </div>
+  )
+}
+```
+
+3. Using [`useCallback`].
+
+```tsx
+const parametricSelector = createSelector(
+  [(state: RootState) => state.todos, (state: RootState, id: number) => id],
+  (todos, id) => todos.filter(todo => todo.id === id)
+)
+
+const MyComponent: FC<Props> = ({ id }) => {
+  const selectTodosById = useCallback(parametricSelector, [])
+
+  const todosById = useSelector(state => selectTodosById(state, id))
+
+  return (
+    <div>
+      {todosById.map(todo => (
+        <div key={todo.id}> {todo.title}</div>
+      ))}
+    </div>
+  )
+}
+```
+
+4. Use [`re-reselect`]:
+
+```ts
+import { createCachedSelector } from 're-reselect'
+
+const parametricSelector = createCachedSelector(
+  [(state: RootState) => state.todos, (state: RootState, id: number) => id],
+  (todos, id) => todos.filter(todo => todo.id === id)
+)((state: RootState, id: number) => id)
+```
+
+Starting in 5.0.0, you can eliminate this problem using `weakMapMemoize`.
+
+###### Using [`createSelector`]
+
+```ts
+const parametricSelector = createSelector(
```

Comment:
:hand: tbh I've never liked the phrase "parametric selector", and I don't think using it as a name in this example helps.  

Also, doing a `.filter()` with _one_ ID in comparison seems unlikely :)  How about filtering by "category" instead?  Seems like something that you might want to do many times.

Reply:
I will update the example, do you want me to name it something generic, what would we call a selector that takes additional parameters? or perhaps we don't actually need to specify that and we can call it something generic "selectSomething"? to stay consistent with the "selectX" pattern?

Reply:
Try to name them meaningfully based on the example (like `selectTodosByCategory`, etc), or just `selectSomething` as a literal fallback, yeah

Reply:
Yeah I will change all the examples to match the `selectX` pattern for consistency.